### ISSUE TITLE
Player inventory gets cleared

### ISSUE THREAD

**Issue by Sparker95** [Sparker95/Vindicta#327]
Reported cases are:
- Interacting with the ACE weapon resting pod
- Giving orders to garrisons
- Recruiting units

So, seems to be quite random, pretty strange.

**Comment by SEBArma** [comment]
might be related to unsupported mods or games being opened via LAN, it never happened to anyone on the "official" server

**Comment by Sparker95** [comment]
So, in one case guy was using these mods:
```
17:24:54                                               name |               modDir |    default |   official |               origin |                                     hash | hashShort | fullPath
17:24:54 ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
17:24:54                                  Enhanced Movement |   @Enhanced Movement |      false |      false |             GAME DIR | fbc1f582c89f11919f9c6ead5d842e6c1aea9b71 |  a47fddc8 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Enhanced Movement
17:24:54                             VET_Unflipping - 1.3.1 |      @VET_Unflipping |      false |      false |             GAME DIR | 415b20dfa474dbd5e43c92dd53daa7df0fcb3793 |  9fcb6e27 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@VET_Unflipping
17:24:54                             Advanced Sling Loading | @Advanced Sling Loading |      false |      false |             GAME DIR | 1af4c6ff7f351b09ad5ef4513f3038d44a375117 |  9f2e7c81 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Advanced Sling Loading
17:24:54                          Advanced Urban Rappelling | @Advanced Urban Rappelling |      false |      false |             GAME DIR | f89771ffb164eb32069350428c73bf5ace17fb12 |  7c4278a8 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Advanced Urban Rappelling
17:24:54                                    Advanced Towing |     @Advanced Towing |      false |      false |             GAME DIR | 3bbdd3d06fbdf74ef0197e92409a34c81edb10d5 |  eb7c2b5d | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Advanced Towing
17:24:54                                Advanced Rappelling | @Advanced Rappelling |      false |      false |             GAME DIR | 35c56daaba7164fbaadb24b5715caf9216189348 |  7029c1d8 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Advanced Rappelling
17:24:54                 Advanced Combat Environment 3.12.6 | @ACE - Legacy Medical Module |      false |      false |             GAME DIR | 51cca4bbb26be976f46cc89ad3d53f1a243bfbcf |  ccf3ed89 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@ACE - Legacy Medical Module
17:24:54                                     Vindicta Addon |    @Vindicta (Alpha) |      false |      false |             GAME DIR | 0adab6cac81d11d1b0eca29a69f8cd9d5f0e906d |  b323d4ae | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@Vindicta (Alpha)
17:24:54                      Community Base Addons v3.13.0 |              @CBA_A3 |      false |      false |             GAME DIR | 61f3ee176e89e625b893e753136f200e70a7d9a6 |  3ecbe430 | D:\SteamLibrary\steamapps\common\Arma 3\!Workshop\@CBA_A3
17:24:54                          Arma 3 Contact (Platform) |                enoch |       true |       true |             GAME DIR | ddd42c4e6f988566cc6139a87da0789d13c90ec1 |  5202fb61 | D:\SteamLibrary\steamapps\common\Arma 3\enoch
17:24:54                                       Arma 3 Tanks |                 tank |       true |       true |             GAME DIR | b89ef4ea37fa57e73ad91ae6970b806b8fd78bea |  b4980166 | D:\SteamLibrary\steamapps\common\Arma 3\tank
17:24:54                                     Arma 3 Tac-Ops |               tacops |       true |       true |             GAME DIR | 58757b6526e7509d21445d5d1b5684974e940e86 |  7816e047 | D:\SteamLibrary\steamapps\common\Arma 3\tacops
17:24:54                                 Arma 3 Laws of War |               orange |       true |       true |             GAME DIR | 15c5640de4f24b45ccf80f1c3c13d2f0fedf9b53 |  cdedb786 | D:\SteamLibrary\steamapps\common\Arma 3\orange
17:24:54                                      Arma 3 Malden |                 argo |       true |       true |             GAME DIR | e728c040496275606583fc76d6e4afd8d501e63a |  38bff3fb | D:\SteamLibrary\steamapps\common\Arma 3\argo
17:24:54                                        Arma 3 Jets |                 jets |       true |       true |             GAME DIR | bd2d51586bc6385591fd4359d060c6b42cfd3e61 |  f48c11b7 | D:\SteamLibrary\steamapps\common\Arma 3\jets
17:24:54                                        Arma 3 Apex |            expansion |       true |       true |             GAME DIR | 03d2bcf543de862e83d1f10e4ebac6ab2f848803 |  a3a32071 | D:\SteamLibrary\steamapps\common\Arma 3\expansion
17:24:54                                    Arma 3 Marksmen |                 mark |       true |       true |             GAME DIR | e9a35bd4235fe6806e63936842da7c420941e0d9 |  e225c0e9 | D:\SteamLibrary\steamapps\common\Arma 3\mark
17:24:54                                 Arma 3 Helicopters |                 heli |       true |       true |             GAME DIR | 11c8e793f4d39cfc7d81c47b2c54aed703b8162a |  8d1e4343 | D:\SteamLibrary\steamapps\common\Arma 3\heli
17:24:54                                       Arma 3 Karts |                 kart |       true |       true |             GAME DIR | fbc65284c714fa51e7914faa618936f867219ee4 |  fd11aba6 | D:\SteamLibrary\steamapps\common\Arma 3\kart
17:24:54                                        Arma 3 Zeus |              curator |       true |       true |             GAME DIR | a22ad2af1877d2b2af7d7c23f55c5f263b271e5a |  59890c95 | D:\SteamLibrary\steamapps\common\Arma 3\curator
17:24:54                                             Arma 3 |                   A3 |       true |       true |            NOT FOUND |                                          |           | 
17:24:54 ==========================================================================================================================================================================================================

```

**Comment by Sparker95** [comment]
I hope with more of those rpts we can find the offending mod easily (hopefully it's not our mod lol)

**Comment by Sparker95** [comment]
Another set of mods:
```
20:34:05                                               name |               modDir |    default |   official |               origin |                                     hash | hashShort | fullPath
20:34:05 ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
20:34:05                                   16AA public mods |   @Interiors for CUP |      false |      false |             GAME DIR | 6293e310b5e6cd063c4adf84443649ae76b5efc8 |  70286545 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Interiors for CUP
20:34:05                 Advanced Diving Environment v0.2.3 | @ADE - Advanced Diving Environment |      false |      false |             GAME DIR | 888ecc690362341b2b7762c7a151c08aa9bca245 |  cf3ecc86 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@ADE - Advanced Diving Environment
20:34:05                        @X-Cam-Taunus (Version 1.1) | @X-Cam-Taunus (Version 1.1) |      false |      false |             GAME DIR | d3c6829f7c80a48b10e94fc118052d4944949324 |   269b6ef | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@X-Cam-Taunus (Version 1.1)
20:34:05                         CUP Terrains - Maps 1.13.0 | @CUP Terrains - Maps |      false |      false |             GAME DIR | f442067187ab1469b25f8d3f3a11d371f8008189 |  9f99815d | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@CUP Terrains - Maps
20:34:05                         CUP Terrains - Core 1.13.0 | @CUP Terrains - Core |      false |      false |             GAME DIR | 47925a60114c38cc42c3c4e2aa5956cf133ffa18 |  e2c189e9 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@CUP Terrains - Core
20:34:05                                     Vindicta Addon |    @Vindicta (Alpha) |      false |      false |             GAME DIR | b84be298000fd963571bd3c21fc3466350b24937 |  67b9efa1 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Vindicta (Alpha)
20:34:05              Advanced Combat Environment 3.13.0.45 | @NEW ACE RHS COMPATIBILITY |      false |      false |             GAME DIR | 12dbfb13e7229124996dcdb36d8a991cde46f643 |  ef1acb34 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@NEW ACE RHS COMPATIBILITY
20:34:05                                          RHS: GREF |             @RHSGREF |      false |      false |             GAME DIR | dcf9f46af60b6abc941c8b5dc06ad031c15eafc4 |  dc7d599d | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@RHSGREF
20:34:05                          RHS: United States Forces |             @RHSUSAF |      false |      false |             GAME DIR | 22a11a80f1287809f28254072472f055d2875c33 |  8d53abc1 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@RHSUSAF
20:34:05        RHS: Armed Forces of the Russian Federation |             @RHSAFRF |      false |      false |             GAME DIR | 0e0e336c4217a33a515de5bb1cffff80f4f0e3db |  e0346bec | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@RHSAFRF
20:34:05                 @L3-GPNVG18 Panoramic Night Vision | @L3-GPNVG18 Panoramic Night Vision |      false |      false |             GAME DIR | 931675607f608b082e873c440374111361d37396 |  36e496d1 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@L3-GPNVG18 Panoramic Night Vision
20:34:05                                Advanced Rappelling | @Advanced Rappelling |      false |      false |             GAME DIR | 35c56daaba7164fbaadb24b5715caf9216189348 |  7029c1d8 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Advanced Rappelling
20:34:05                                   Achilles (1.2.2) |            @Achilles |      false |      false |             GAME DIR | 233199092d6643240eb75db9d8e7ddb50c812a2d |  71d00e09 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Achilles
20:34:05                  Task Force Arrowhead Radio 0.9.12 |    @task_force_radio |      false |      false |             GAME DIR | 3028ba91015d709af11bfe12a4c851569d259050 |  ebed631c | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@task_force_radio
20:34:05                            ShackTac User Interface | @ShackTac User Interface |      false |      false |             GAME DIR | 6017b07d0a32d8dfb2015dd952868b5764015fd3 |  ab7edfbc | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@ShackTac User Interface
20:34:05                 Advanced Combat Environment 3.13.0 |                 @ace |      false |      false |             GAME DIR | da39a3ee5e6b4b0d3255bfef95601890afd80709 |  11fdd19c | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@ace
20:34:05                         JSRS Soundmod - CE.19.1117 |       @JSRS SOUNDMOD |      false |      false |             GAME DIR | 87cdbb7390ad14b9ed8ea07ef8ab79228cce8352 |  ca7f8ae3 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@JSRS SOUNDMOD
20:34:05                                  Enhanced Movement |   @Enhanced Movement |      false |      false |             GAME DIR | fbc1f582c89f11919f9c6ead5d842e6c1aea9b71 |  a47fddc8 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Enhanced Movement
20:34:05             @Blastcore Edited (standalone version) | @Blastcore Edited (standalone version) |      false |      false |             GAME DIR | 41dd8517557c087270ce49c5381e5d8224ebd150 |  9c77b2da | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@Blastcore Edited (standalone version)
20:34:05                      Community Base Addons v3.13.0 |              @CBA_A3 |      false |      false |             GAME DIR | 61f3ee176e89e625b893e753136f200e70a7d9a6 |  3ecbe430 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@CBA_A3
20:34:05                                  @@Swiss_Divisions |    @@Swiss_Divisions |      false |      false |             GAME DIR | ad7a5771cf2d136aea1828cb3ebfaec6b455b7ea |  962a5423 | G:\SteamGames\steamapps\common\Arma 3\!Workshop\@@Swiss_Divisions
20:34:05                          Arma 3 Contact (Platform) |                enoch |       true |       true |             GAME DIR | ddd42c4e6f988566cc6139a87da0789d13c90ec1 |  5202fb61 | G:\SteamGames\steamapps\common\Arma 3\enoch
20:34:05                                       Arma 3 Tanks |                 tank |       true |       true |             GAME DIR | b89ef4ea37fa57e73ad91ae6970b806b8fd78bea |  b4980166 | G:\SteamGames\steamapps\common\Arma 3\tank
20:34:05                                     Arma 3 Tac-Ops |               tacops |       true |       true |             GAME DIR | 58757b6526e7509d21445d5d1b5684974e940e86 |  7816e047 | G:\SteamGames\steamapps\common\Arma 3\tacops
20:34:05                                 Arma 3 Laws of War |               orange |       true |       true |             GAME DIR | 15c5640de4f24b45ccf80f1c3c13d2f0fedf9b53 |  cdedb786 | G:\SteamGames\steamapps\common\Arma 3\orange
20:34:05                                      Arma 3 Malden |                 argo |       true |       true |             GAME DIR | e728c040496275606583fc76d6e4afd8d501e63a |  38bff3fb | G:\SteamGames\steamapps\common\Arma 3\argo
20:34:05                                        Arma 3 Jets |                 jets |       true |       true |             GAME DIR | bd2d51586bc6385591fd4359d060c6b42cfd3e61 |  f48c11b7 | G:\SteamGames\steamapps\common\Arma 3\jets
20:34:05                                        Arma 3 Apex |            expansion |       true |       true |             GAME DIR | 03d2bcf543de862e83d1f10e4ebac6ab2f848803 |  a3a32071 | G:\SteamGames\steamapps\common\Arma 3\expansion
20:34:05                                    Arma 3 Marksmen |                 mark |       true |       true |             GAME DIR | e9a35bd4235fe6806e63936842da7c420941e0d9 |  e225c0e9 | G:\SteamGames\steamapps\common\Arma 3\mark
20:34:05                                 Arma 3 Helicopters |                 heli |       true |       true |             GAME DIR | 11c8e793f4d39cfc7d81c47b2c54aed703b8162a |  8d1e4343 | G:\SteamGames\steamapps\common\Arma 3\heli
20:34:05                                       Arma 3 Karts |                 kart |       true |       true |             GAME DIR | fbc65284c714fa51e7914faa618936f867219ee4 |  fd11aba6 | G:\SteamGames\steamapps\common\Arma 3\kart
20:34:05                                        Arma 3 Zeus |              curator |       true |       true |             GAME DIR | a22ad2af1877d2b2af7d7c23f55c5f263b271e5a |  59890c95 | G:\SteamGames\steamapps\common\Arma 3\curator
20:34:05                                             Arma 3 |                   A3 |       true |       true |            NOT FOUND |                                          |           | 
20:34:05 ==========================================================================================================================================================================================================

```

**Comment by Sparker95** [comment]
another mod list from Night Shade:
```
ch view distance
dynamic view distance
enhanced movement
```

**Comment by ChlorideCull** [comment]
Is this what happened to me, or is there another inventory bug? Clothes content and backpack content was cleared, but it didn't remove the clothes or backpack.

I was running it in single player, and it occurred while running from camp to a nearby town. [rpt here](https://gist.github.com/ChlorideCull/1d911659052f1b13a869dc332b887a08) - sad to say that I'm just running some DLC, CBA, and ACE, so any issue wouldn't be a conflict.

**Comment by Sparker95** [comment]
Can not tell exactly, but appears to be the same kind of bug. Very weird.

**Comment by marvis999** [comment]
Some things people have said:
+ It happened using the first slot of the role selection (multiple people)
+ It possibly happens only on locally hosted servers, not dedicated (multiple people)
+ One person said it happened in Antistasi too
+ Recruiting seems to commonly cause it

It probably has something to do with Jeroen's arsenal and the way it removes items from it.

**Comment by marvis999** [comment]
Video evidence:
https://www.twitch.tv/videos/547900808?t=00h47m27s
https://www.twitch.tv/videos/547900808?t=00h49m23s

Seems to happen very randomly.
It removes:
- Uniform, vest and backpack content
- All weapons
- But not the uniform, vest or backpack themselves

**Comment by Jeroen-Notenbomer** [comment]
I think this guy is on discord KeplerTv#4519 
incase we need to ask him something about his games setup

**Comment by marvis999** [comment]
Yes that is more likely the cause. 

I couldn't find any other report about it for Antistasi.

**Comment by Jeroen-Notenbomer** [comment]
loadInventory is only called localy in my arsenal code. So i dont see a way it can be called when you are not in the arsenal yourself.

It also doesnt seem to remove other items like map, radio bino or nv. So that is another indication its not the loadinventory script.

**Comment by billw2012** [comment]
> Some possible candidate:
> https://github.com/Sparker95/Vindicta/blob/d73d3074f611c3c7ab301266f66565c0a165dbd8/Vindicta.Altis/Unit/Unit.sqf#L1170

I did think this, however on looking again you can clearly see it also removes the vest, and we know that isn't part of the bug due to the video above. It is ONLY clearing sub containers and removing weapons, it leaves backpack, outfit, vest and gadgets.

**Comment by Jeroen-Notenbomer** [comment]
https://github.com/Sparker95/Vindicta/blob/d73d3074f611c3c7ab301266f66565c0a165dbd8/Vindicta.Altis/Unit/Unit.sqf#L1176

Its a bit of a strange construction but it adds it back directly after if it had one already

**Comment by billw2012** [comment]
yeah ignore me i didn't read it properly!

**Comment by nemosaint** [comment]
I have this bug every time in the steam workshop version. Happens also multiple times in one session. Seems to be random. Is this solved in the newer versions? Or still a problem/mystery I host server on my own PC (not dedicated) and I'm the only one affected by this. 
We play with CBA_A3, TFAR, ACE, Vindicta (Alpha).

**Comment by billw2012** [comment]
Should be fixed in 0.36
